pkg/core/core.go

```diff
@@ -117,6 +120,18 @@ func (c *Core) Hook(ctx context.Context, id uint64, _ models.HookOptions) error
 			utils.LogError(c.logger, err, "failed to unload the hooks")
 		}
 
+		if opts.Mode != models.MODE_TEST {
```

Comment:
you can use 
```
if opts.Mode == models.MODE_TEST  &&  c.hostConfigStr != "" {
}
```
current thing is a bit hard to read

Reply:
cool